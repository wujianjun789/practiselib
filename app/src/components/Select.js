/**
 * Created by a on 2017/7/7.
 */
import React, { Component } from 'react'
import Immutable from 'immutable';

export default class Select extends Component {
    constructor(props) {
        super(props)
        this.onChange = this.onChange.bind(this);
    }

    onChange(event) {
        this.props.onChange && this.props.onChange(event.target.selectedIndex);
    }

    render() {
        const {className='', id='', valueKey="value", titleKey="value", data=Immutable.fromJS({
                list: [],
                value: '',
                placeholder: ""
            })} = this.props;

        return <select className={ "select " + className + " form-control" } id={ id } placeholder={ data.get('placeholder') } value={ data.get("value") } onChange={ this.onChange }>
                 { data.get('list').map((option, index) => {
                       let value = option.get(valueKey);
                       let title = option.get(titleKey);
                       return <option key={ index } value={ value }>
                                { title }
                              </option>
                   }) }
               </select>
    }
}