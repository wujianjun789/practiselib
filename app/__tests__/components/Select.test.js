/**
 * Created by a on 2017/7/21.
 */
import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import Select from '../../src/components/Select';

import Immutable from 'immutable';

const data = Immutable.fromJS({list:[{id:1, value:'域'},{id:2, value:'域2'}], index:0, value:'域'})
test('SidebarInfo renders', ()=>{

    const component = renderer.create(
        <Select data={data}/>
    )

    let select = component.toJSON();
    expect(select).toMatchSnapshot();
})

test('Select onChange', ()=>{
    const onChange = jest.fn();
    const component = shallow(<Select data={data} onChange={onChange}/>)

    component.simulate('change');
    expect(onChange).toHaveBeenCalledTimes(1);
})