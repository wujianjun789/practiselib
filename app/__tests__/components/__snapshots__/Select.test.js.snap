exports[`test Select renders 1`] = `
<select
  className="select  form-control"
  id=""
  onChange={[Function]}
  placeholder={undefined}
  value="域">
  <option
    value="域">
    域
  </option>
  <option
    value="域2">
    域2
  </option>
</select>
`;
